crmd/lrm_state.c

```diff
@@ -222,6 +229,11 @@ lrm_state_reset_tables(lrm_state_t * lrm_state)
                   g_hash_table_size(lrm_state->rsc_info_cache));
         g_hash_table_remove_all(lrm_state->rsc_info_cache);
     }
+    if (lrm_state->metadata_cache) {
+        crm_trace("Re-setting metadata cache with %d members",
+                  g_hash_table_size(lrm_state->metadata_cache));
+        g_hash_table_remove_all(lrm_state->metadata_cache);
+    }
```

Comment:
Since _crmd/lrmd_state.c:lrm_state_reset_tables()_ is already invoked in _crmd/remote_lrmd_ra.c:remote_lrm_op_callback()_ on start and migrate_from actions, I think we only need to save the metadata cache when the tables get reset in _crmd/remote_lrmd_ra.c:handle_remote_ra_stop()_ with something like
```
metadata_hash_save = lrm_state->metadata_hash;
lrm_state->metadata_hash = NULL;
lrm_state_reset_tables(lrm_state);
lrm_state->metadata_hash = metadata_hash_save;
```
Did I get it right?